ui/src/message_popup/renderers/audio_response_summary.jsx

```diff
@@ -0,0 +1,94 @@
+/* @flow weak */
+import _ from 'lodash';
+import React from 'react';
+import VelocityTransitionGroup from "velocity-react/velocity-transition-group";
+import Divider from 'material-ui/Divider';
+
+import ReadMore from './read_more.jsx';
+
+/*
+Component that displays the summary of audio responses.
+*/
+export default React.createClass({
+  displayName: 'AudioResponseSummary',
+
+  propTypes: {
+    gameSession: React.PropTypes.object.isRequired,
+  },
+
+  render() {
+    const audioResponses = [];
+    const reviewedQuestions = {};  // Keeps track of reviewedQuestions to ensure we only show the last response if the user does multiple retries.
+    this.props.gameSession.audioResponses.map((arObj, i) => {
+      var obj = {};
+      obj.audioUrl = arObj.blobUrl;
+      obj.questionText = arObj.question.text;
+      var questionId = arObj.question.id;
+      if(questionId in reviewedQuestions) {
+        audioResponses[reviewedQuestions[questionId]] = obj;
+      } else {
+        reviewedQuestions[questionId] = audioResponses.length;
+        audioResponses.push(obj);
+      }
+    });
+
+    return (
+      <div className="done">
+        <VelocityTransitionGroup enter={{animation: "slideDown"}} leave={{animation: "slideUp"}} runOnMount={true}>
+          <div style={styles.doneTitle}>
+            <p style={styles.paragraph}>You've finished the simulation.</p>
+            <p style={styles.paragraph}><strong>Do not close this page</strong>. You will need it for the reflection.</p>
+            <p style={styles.paragraph}>Please return to the form for the post-simulation reflection.</p>
+          </div>
+          <p style={styles.summaryTitle}>Summary</p>
+          <Divider />
+          {audioResponses.map((obj, i) =>
+            <div key={i} style ={_.merge(styles.instructions)}>
```

Comment:
A better key here is the `questionId`.  Also can you please use a real variable name like `audioResponse` in place of `obj`?